### PR TITLE
SAIC-583 added creating of zero image in generate_load

### PR DESCRIPTION
**Pull request by ntymtsiv**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
MirantisWorkloadMobility

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.generate_load
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/generate_load.py:36: [W0621(redefined-outer-name), clean_if_exists.wrapper] Redefining name 'args' from outer scope (line 1172)
devlab/tests/generate_load.py:56: [W0612(unused-variable), retry_until_resources_created.actual_decorator.wrapper] Unused variable 'i'
devlab/tests/generate_load.py:490: [W0612(unused-variable), Prerequisites.create_vms.wait_for_vm_creating] Unused variable 'i'
devlab/tests/generate_load.py:645: [W0612(unused-variable), Prerequisites.create_all_networking] Unused variable 'i'
devlab/tests/generate_load.py:771: [W0612(unused-variable), Prerequisites.generate_vm_state_list] Unused variable 'i'
devlab/tests/generate_load.py:791: [W0703(broad-except), Prerequisites.delete_flavor] Catching too general exception Exception
devlab/tests/generate_load.py:953: [W0120(useless-else-on-loop), CleanEnv.clean_vms.wait_until_vms_all_deleted] Else clause on loop without a break statement
devlab/tests/generate_load.py:943: [W0612(unused-variable), CleanEnv.clean_vms.wait_until_vms_all_deleted] Unused variable 'i'
devlab/tests/generate_load.py:961: [W0106(expression-not-assigned), CleanEnv.clean_vms] Expression "[vms.append(vm) for vm in self.config.vms_from_volumes]" is assigned to nothing
devlab/tests/generate_load.py:1071: [W0703(broad-except), CleanEnv.clean_fips] Catching too general exception Exception
devlab/tests/generate_load.py:1106: [W0141(bad-builtin), CleanEnv.clean_keypairs.delete_user_keypairs] Used builtin function 'map'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1398/

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Grizzly to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-juno/608/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

File: devlab/tests/generate_load.py [hunk 1]
```diff
@@ -403,6 +403,8 @@ def wait_until_images_created(image_ids):
                         member_id = tenant['id']
                         self.glanceclient.image_members.create(image_id,
                                                                member_id)
+        if getattr(self.config, 'create_zero_image'):
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]: Why not simply `self.config.create_zero_image`?

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.generate_load
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/generate_load.py:36: [W0621(redefined-outer-name), clean_if_exists.wrapper] Redefining name 'args' from outer scope (line 1172)
devlab/tests/generate_load.py:56: [W0612(unused-variable), retry_until_resources_created.actual_decorator.wrapper] Unused variable 'i'
devlab/tests/generate_load.py:490: [W0612(unused-variable), Prerequisites.create_vms.wait_for_vm_creating] Unused variable 'i'
devlab/tests/generate_load.py:645: [W0612(unused-variable), Prerequisites.create_all_networking] Unused variable 'i'
devlab/tests/generate_load.py:771: [W0612(unused-variable), Prerequisites.generate_vm_state_list] Unused variable 'i'
devlab/tests/generate_load.py:791: [W0703(broad-except), Prerequisites.delete_flavor] Catching too general exception Exception
devlab/tests/generate_load.py:953: [W0120(useless-else-on-loop), CleanEnv.clean_vms.wait_until_vms_all_deleted] Else clause on loop without a break statement
devlab/tests/generate_load.py:943: [W0612(unused-variable), CleanEnv.clean_vms.wait_until_vms_all_deleted] Unused variable 'i'
devlab/tests/generate_load.py:961: [W0106(expression-not-assigned), CleanEnv.clean_vms] Expression "[vms.append(vm) for vm in self.config.vms_from_volumes]" is assigned to nothing
devlab/tests/generate_load.py:1071: [W0703(broad-except), CleanEnv.clean_fips] Catching too general exception Exception
devlab/tests/generate_load.py:1106: [W0141(bad-builtin), CleanEnv.clean_keypairs.delete_user_keypairs] Used builtin function 'map'

File: devlab/tests/generate_load.py [hunk 2]
```diff
@@ -403,6 +403,8 @@ def wait_until_images_created(image_ids):
                         member_id = tenant['id']
                         self.glanceclient.image_members.create(image_id,
                                                                member_id)
+        if getattr(self.config, 'create_zero_image', None):
```
Comment (ntymtsiv) [inline devlab/tests/generate_load.py]: getattr is used instead of self.config.create_zero_image to make create_zero_image  parameter optional in config.py

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1409/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by plomakin** [comment]
+1 works for me

**Comment by roman-verchikov** [comment]
Ok, +1, can be merged.

**Pull request merged**